packages/grid/x-grid/src/tests/pagination.XGrid.test.tsx

```diff
@@ -7,51 +7,92 @@ import { getColumnValues } from 'test/utils/helperFn';
 import * as React from 'react';
 import { expect } from 'chai';
 import { XGrid, useGridApiRef } from '@material-ui/x-grid';
+import { useData } from 'packages/storybook/src/hooks/useData';
 
 const isJSDOM = /jsdom/.test(window.navigator.userAgent);
 
 describe('<XGrid /> - Pagination', () => {
   // TODO v5: replace with createClientRender
   const render = createClientRenderStrictMode();
 
-  const baselineProps = {
-    autoHeight: isJSDOM,
-    rows: [
-      {
-        id: 0,
-        brand: 'Nike',
-      },
-      {
-        id: 1,
-        brand: 'Adidas',
-      },
-      {
-        id: 2,
-        brand: 'Puma',
-      },
-    ],
-    columns: [{ field: 'brand', width: 100 }],
-  };
-
-  it('should apply setPage correctly', () => {
-    let apiRef;
-    const GridTest = () => {
-      apiRef = useGridApiRef();
-
-      return (
-        <div style={{ width: 300, height: 300 }}>
-          <XGrid {...baselineProps} apiRef={apiRef} pagination pageSize={1} hideFooter />
-        </div>
-      );
-    };
-
-    render(<GridTest />);
-
-    expect(getColumnValues()).to.deep.equal(['Nike']);
-    act(() => {
-      apiRef.current.setPage(1);
+  describe('setPage', () => {
+    it('should apply valid value', () => {
+      let apiRef;
+      const GridTest = () => {
+        const basicData = useData(20, 2);
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid {...basicData} apiRef={apiRef} pagination pageSize={1} />
+          </div>
+        );
+      };
+
+      render(<GridTest />);
+
+      expect(getColumnValues()).to.deep.equal(['0']);
+      act(() => {
+        apiRef.current.setPage(1);
```

Comment:
Interesting, tests with `setPage` don't need `act` but `setPageSize` won't pass without it.